BUILD

```diff
@@ -2712,7 +2712,7 @@ grpc_cc_library(
     ],
     external_deps = [
         "upb_lib",
-        "upb_lib_descriptor",
+        "upb_lib_descriptor_reflection",
```

Comment:
I'm puzzled by this: Can you elaborate a bit on why replacing `upb_lib_descriptor` with `upb_lib_descriptor_reflection` does anything when the conflict we're seeing is "undeclared dependency on descriptor.upbdefs.h"? (it seems like the _reflection version of the library would also depdend on descriptor.upbdefs.h and descriptor.upbdefs.c)

Reply:
These replacements are not meant to fix the error. They're merely updated because they're supposed to rely upon it. (e.g. `envoy_ads_upbdefs` should depend on `upb_lib_descriptor_reflection` because it's a collection of upb reflections (`*.upbdefs`) and it has to rely on the reflection of descriptor (`upb_lib_descriptor_reflection`).

Actually I don't know why this PR happens to fix the original error, to be honest. I think the gRPC build should work without this PR but it happen not to work sometimes on Windows. My guess is that the upb project has  `descriptor.upbdefs.*` which will be generated by one of its targets and gRPC also has the same file name in its project. Mystery here is gRPC doesn't actually depend on the upb target which generates those files. (gRPC just embedded it instead of depending on it) But bazel on Windows appears to be confused by this.

Therefore, I just makes gRPC depend on the target instead of embedding it, which seems to work. :)